src/config/config_update.c

```diff
@@ -82,8 +83,9 @@ static int refactor_entry(char* src_p, char* dest_p, const char* const key_p, co
     skip_space(&pos_p);
 
     /* found "key = ", copy in the new value */
-    int val_len = strnlen(val_p, max_dest_len);
-    line_len = pos_p - src_p;
+    const size_t val_len = strnlen(val_p, max_dest_len);
+    assert(pos_p >= src_p);
+    line_len = (size_t)(pos_p - src_p);
```

Comment:
Also here, I suggest replacing the new assertion by, e.g., `/* note that pos_p >= src_p */` at the end of the following line
and removing the `#include <assert.h>`.